manifests/server.pp

```diff
@@ -465,12 +470,16 @@
   $use_legacy_auth_conf      = $::puppet::server_use_legacy_auth_conf,
 ) {
 
+  if $server_facts != $server_foreman_facts {
```

Comment:
This will match when `server_foreman_facts` is changed, it'll force people to set the old parameter too, in order to prevent the warning. Perhaps set the old parameter default to `undef` and prefer it if set, or even remove it entirely (we can do a major release next, perhaps)?